demo/lightning/checkpoint/simulated/multiprocessing_train.py

```diff
@@ -15,24 +15,120 @@
  """
 import argparse
 import os
-import time
 import statistics
-from typing import Optional, Dict
+import time
+from typing import Dict, Optional, TextIO
+
 
 import torch
-import torch.nn as nn
 import torch.distributed as dist
-import torch.multiprocessing as mp
 import torch.distributed.checkpoint as dist_cp
+import torch.multiprocessing as mp
+import torch.nn as nn
+
 from lightning.pytorch.strategies import FSDPStrategy
-from dataflux_pytorch.lightning.gcs_filesystem import GCSDistributedWriter, GCSDistributedReader
 
+from dataflux_pytorch.lightning.gcs_filesystem import GCSDistributedReader, GCSDistributedWriter
+
+# Constants for distributed setup
 MASTER_ADDR = 'localhost'
 MASTER_PORT = '12355'
 
 
+def write_model_structure(file: TextIO, model: nn.Module, indent: str = '') -> None:
```

Comment:
I would specifiy `_to_file` in the name to disambiguate from writing to the actual model itself.

Reply:
Done